notebook/terminal/handlers.py

```diff
@@ -14,8 +15,31 @@ class TerminalHandler(IPythonHandler):
     """Render the terminal interface."""
     @web.authenticated
     def get(self, term_name):
-        self.write(self.render_template('terminal.html',
-                   ws_path="terminals/websocket/%s" % term_name))
+        if term_name == 'new':
+            model = self.terminal_manager.create()
+            term_name = model['name']
+            new_path = self.request.path.replace("terminals/new", "terminals/" + term_name)
+            self.redirect(new_path)
```

Comment:
Good approach. It might be good to make this an explicit path in the handlers above with `(ujoin(base_url, r"/terminals/new"), ReallyActuallyNewTerminalHandler),`, noting that you may have to make it the first entry in the list (and figuring out a better name for that).